docs/using/EXTENDING_DOCKER.adoc

```diff
@@ -47,8 +47,58 @@ See the https://github.com/jenkinsci/plugin-installation-manager-tool[Plugin Ins
 
 == Adding tools
 
-NOTE: This documentation section has not been implemented yet.
-Please feel free to submit a pull request!
+To add a tool let's say `git`, you can use https://github.com/jenkinsci/configuration-as-code-plugin[Jenkins Configuration-as-Code]. To know more about tools in jenkins, see https://www.jenkins.io/doc/book/managing/tools/[Managing Tools].
+
+To configure a tool in jenkins, you first need to install required plugins for that tool, so you can install plugins in `jenkinsfile-runner` as shown in link:#_installing_plugins[Installing Plugins] section. As `git` plugin is already present in the `jenkins/jenkinsfile-runner` docker image, so no need to install it again.
+
+*Note:* Here `git` tool is different than `git` plugin, here `git` tool is the `git` command that jenkins internally uses to clone repositories.
+
+Most of the tool plugins in jenkins has support to automatically download tool from internet and install it which can be specified in Jenkins CasC, like shown in this https://github.com/jenkinsci/configuration-as-code-plugin/tree/master/demos/nodejs[example]. But If you want to manually install the tool, you can write the installation steps of the tool in the same Dockerfile (that used to install plugins), like here for `git`, the preferred method to install `git` command is with OS Package Manager.
+
+Dockerfile::
+
+[source]
+----
+FROM jenkins/jenkinsfile-runner
+RUN apt-get update && apt-get install -y git && rm -rf /var/lib/apt/lists/*
+----
+
+Next, Creating Jenkins CasC YAML file for configuring `git` tool in jenkins. Refer to link:/demo/casc/README.md[JCasC demo] for more examples.
+
+jenkins.yaml::
+
+[source]
+----
+tool:
+  git:
+    installations:
+      - name: git
+        home: /usr/bin/git
+----
```

Comment:
Here we are installing git in jenkinsfile-runner docker image which is linux based, so I think there is no need to specify the installation paths for macOS or windows users.

Reply:
You will have to explain at least something like "the jenkinsfile-runner docker image is linux-based", or people will not be aware of it.

Reply:
I think the one who knows jenkins, can easily understand this, and also isn't it obvious that it is Linux based docker image for anyone using jenkinsfile-runner docker image?

Reply:
Well we have to consider new-comers to Jenkins as well like yourself.

Reply:
yeah, you are right but see all docker images created from base images like ubuntu, centos and these are linux based and since here this is the Docker doc for jenkinsfile-runner, so I guess there is no need to mention anything like here.